src/mongoose_scram.erl

```diff
@@ -164,6 +207,11 @@ deserialize(Bin) ->
     ?WARNING_MSG("Corrupted serialized SCRAM: ~p", [Bin]),
     {error, corrupted_scram}.
 
+deserialize([], _) ->
+    [];
+deserialize([Sha| RemainingSha] , [ _Prefix, StoredKey, ServerKey | RemainingKeys] ) ->
```

Comment:
Yep, the deserialization surely is not there yet. ejabberd_auth_internal is not using it hence I've put this here as a draft. I was also thinking of making it more flexible in case of adding/removing sha. 
What do you think if supported_sha_types would return list of tuples {sha_atom, <<"==SHA_PREFIX==">>} and while iterating we'd know which Prefix to expect and assign to correct map element.

Reply:
It can return such a list and when iterating over the `DeserializedKey` we check if the Prefix is on the `supported_sha_types` list. Just don't make any assumptions on the order of the sha version in the `DeserializedKey` list.